c_src/ep_decoder.c

```diff
@@ -896,12 +896,19 @@ fill_default(ErlNifEnv *env, ep_spot_t *spot)
 
             case field_float:
             case field_double:
-                *t++ = state->double_zero;
+                *t++ = enif_make_double(env, 0.0);
                 break;
 
             case field_string:
             case field_bytes:
-                *t++ = state->binary_nil;
+                {
+                     ErlNifBinary bin;
+                     if (!enif_alloc_binary(0, &bin)) {
```

Comment:
why would this call fail and the one in else succeed?

Reply:
the function returns true on success - so if we successfully allocated, we'll go to the else branch. agreed that this is kind of confusing but i'm following the style on line 937.